include/open62541/server_pubsub.h

```diff
@@ -138,18 +138,26 @@ typedef enum  {
 } UA_PubSubComponentEnumType;
 
 typedef enum {
-    UA_PUBSUB_PUBLISHERID_NUMERIC,
-    UA_PUBSUB_PUBLISHERID_STRING
+    UA_PUBLISHERIDTYPE_BYTE = 0,
+    UA_PUBLISHERIDTYPE_UINT16 = 1,
+    UA_PUBLISHERIDTYPE_UINT32 = 2,
+    UA_PUBLISHERIDTYPE_UINT64 = 3,
+    UA_PUBLISHERIDTYPE_STRING = 4
 } UA_PublisherIdType;
 
+typedef union { /* std: valid types UInt or String */
```

Comment:
The comment is a bit misleading.
All UInt types are permitted.

Make a direct reference to Part 14, 7.2.2.2.2 NetworkMessage Layout:

> Bit range 0-2: PublisherId Type
000 The PublisherId is of DataType Byte This is the default value if ExtendedFlags1 is omitted
001 The PublisherId is of DataType UInt16
010 The PublisherId is of DataType UInt32
011 The PublisherId is of DataType UInt64
100 The PublisherId is of DataType String

Reply:
Extended comment